app/src/ui/banners/banner.tsx

```diff
@@ -4,49 +4,64 @@ import * as OcticonSymbol from '../octicons/octicons.generated'
 
 interface IBannerProps {
   readonly id?: string
-  readonly timeout?: number
   readonly dismissable?: boolean
   readonly onDismissed: () => void
 }
 
 export class Banner extends React.Component<IBannerProps, {}> {
-  private timeoutId: number | null = null
+  private contents = React.createRef<HTMLDivElement>()
+  private closeButton = React.createRef<HTMLButtonElement>()
+  private focusTimeoutId: number | null = null
 
   public render() {
     return (
-      <div id={this.props.id} className="banner">
-        <div className="contents">{this.props.children}</div>
+      <div
+        id={this.props.id}
+        className="banner"
+        aria-atomic="true"
+        role="alert"
+      >
+        <div className="contents" ref={this.contents}>
+          {this.props.children}
+        </div>
         {this.renderCloseButton()}
       </div>
     )
   }
 
   private renderCloseButton() {
     const { dismissable } = this.props
-    if (dismissable === undefined || dismissable === false) {
+
+    if (dismissable === false) {
       return null
     }
 
     return (
       <div className="close">
-        <button onClick={this.props.onDismissed}>
+        <button
+          onClick={this.props.onDismissed}
+          aria-label="Dismiss this message"
+          ref={this.closeButton}
+        >
           <Octicon symbol={OcticonSymbol.x} />
         </button>
       </div>
     )
   }
 
-  public componentDidMount = () => {
-    if (this.props.timeout !== undefined) {
-      this.timeoutId = window.setTimeout(() => {
-        this.props.onDismissed()
-      }, this.props.timeout)
-    }
+  public componentDidMount(): void {
+    this.focusTimeoutId = window.setTimeout(() => {
+      if (this.closeButton.current) {
+        this.closeButton.current.focus()
+      } else {
+        this.contents.current?.querySelector('a')?.focus()
```

Comment:
I believe we should have this inverted in that if there is a link in the contents, that gets focused and if not then focus the dismiss button. This is the pattern we follow on our dialogs as well and has roots in the accessibility concept of "focus the first interactive element". 

Other thoughts are to pull this into a method like in the `dialog.tsx` and use the `sleep` helper method to make this code a little easier to read and could add some documentation - especially explaning the need for the timeout. Something like: 
```
  private async focusFirstSuitableElement() {
     // Timeout is required because flows such as rebase are in dialogs and have focus management that sets the focus to the app body after this has run.
    await sleep(200)
    
     const link = this.contents.current?.querySelector('a')
     if (link) {
      link.focus()
      return
    }
    
    this.closeButton.current?.focus()
  }
```

Reply:
```suggestion
      if (this.closeButton.current) {
        this.closeButton.current.focus()
      } else {
        this.contents.current?.querySelector('a')?.focus()
```

Reply:
@tidy-dev I've looked at the implementation of `sleep`, and I think the only issue might be that there is no simple way to hold onto the ID's that come from the `window.setTimeout` calls. Right now AFAICT we make sure to clean up our timeouts to avoid calling `onDismiss` or moving focus to non-existent elements if the component is unmounted.